txn_test.go

```diff
@@ -259,6 +260,130 @@ func TestIndexed(t *testing.T) {
 	})
 }
 
+func TestSortIndex(t *testing.T) {
+	c := NewCollection()
+	c.CreateColumn("col1", ForString())
+	c.CreateSortIndex("sortedCol1", "col1")
+
+	assert.Error(t, c.CreateSortIndex("", ""))
+	assert.Error(t, c.CreateSortIndex("no_col", "nonexistent"))
+	assert.Error(t, c.CreateSortIndex("sortedCol1", "col1"))
+
+	// Inserts
+	c.Insert(func (r Row) error {
+		r.SetString("col1", "bob")
+		return nil
+	})
+	c.Insert(func (r Row) error {
+		r.SetString("col1", "carter")
+		return nil
+	})
+	c.Insert(func (r Row) error {
+		r.SetString("col1", "dan")
+		return nil
+	})
+	c.Insert(func (r Row) error {
+		r.SetString("col1", "alice")
+		return nil
+	})
+	
+	// Update
+	assert.NoError(t, c.QueryAt(3, func(r Row) error {
+		r.SetString("col1", "rob")
+		return nil
+	}))
+	indexCol, _ := c.cols.Load("sortedCol1")
+	assert.Equal(t, 4, indexCol.Column.(*columnSortIndex).btree.Len())
+	
+	// Delete
+	assert.Equal(t, true, c.DeleteAt(1))
+	assert.Equal(t, 3, indexCol.Column.(*columnSortIndex).btree.Len())
+
+	// Range
+	assert.Error(t, c.Query(func (txn *Txn) error {
+		return txn.SortedRange("nonexistent", func (i uint32) {
+			return
+		})
+	}))
+
+	var res [3]string
+	var resN int = 0
+	c.Query(func (txn *Txn) error {
+		col1 := txn.String("col1")
+		return txn.SortedRange("sortedCol1", func (i uint32) {
+			name, _ := col1.Get()
+			res[resN] = name
+			resN++
+		})
+	})
+
+	assert.Equal(t, "bob", res[0])
+	assert.Equal(t, "dan", res[1])
+	assert.Equal(t, "rob", res[2])
+}
+
+func TestSortIndexLoad(t *testing.T) {
+
+	players := loadPlayers(500)
+	players.CreateSortIndex("sorted_names", "name")
+
+	checkN := 0
+	checks := map[int]string{
+		4: "Buckner Frazier",
+		16: "Marla Todd",
+		30: "Shelly Kirk",
+		35: "out of range",
+	}
+
+	players.Query(func (txn *Txn) error {
+		txn = txn.With("human", "mage")
+		name := txn.String("name")
+		txn.SortedRange("sorted_names", func (i uint32) {
+			n, _ := name.Get()
+			if res, exists := checks[checkN]; exists {
+				assert.Equal(t, res, n)
+			}
+			checkN++
+		})
+		return nil
+	})
+
+}
+
+func TestSortIndexChunks(t *testing.T) {
+	N := 100_000
+	obj := map[string]any{
+		"name": "1",
+		"balance": 12.5,
+	}
+
+	players := NewCollection()
+	players.CreateColumnsOf(obj)
+	players.CreateSortIndex("sorted_names", "name")
+
+	for i := 0; i < N; i++ {
+		players.Insert(func (r Row) error {
+			return r.SetMany(map[string]any{
+				"name": strconv.Itoa(i),
+				"balance": float64(i) + 0.5,
+			})
+		})
+	}
+
+	players.Query(func (txn *Txn) error {
+		name := txn.String("name")
+		txn.SortedRange("sorted_names", func (i uint32) {
+			n, _ := name.Get()
+			if i % 400 == 0 {
+				nInt, _ := strconv.Atoi(n)
+				assert.Equal(t, nInt, int(i))
+			}
+		})
+		return nil
+	})
+}
+
```

Comment:
We probably need a test to check for concurrency, when you update b-tree while ranging over it.